sound/soc/sof/sof-client-probes.c

```diff
@@ -430,7 +416,22 @@ static int sof_probes_client_probe(struct auxiliary_device *auxdev,
 	}
 
 	priv->host_ops = ops;
-	priv->ipc_ops = &ipc3_probe_ops;
+
+	if (sof_client_get_ipc_type(cdev) == SOF_INTEL_IPC4) {
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_IPC4)
+		priv->ipc_ops = &ipc4_probe_ops;
+#endif
+	} else if (sof_client_get_ipc_type(cdev) == SOF_IPC) {
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_IPC3)
+		priv->ipc_ops = &ipc3_probe_ops;
+#endif
```

Comment:
same, SOF_IPC means IPC3. Can we simplify?

Reply:
Sure.